modules/indexing/src/test/java/org/apache/ignite/util/GridCommandHandlerInterruptCommandTest.java

```diff
@@ -0,0 +1,258 @@
+            this.y = y;
+            this.z = z;
+        }
+
+        /**
+         * @param seed Seed.
+         */
+        public UserValue(long seed) {
+            x = (int)(seed % 6991);
+            y = (int)(seed % 18679);
+            z = (int)(seed % 31721);
+        }
+
+        /** {@inheritDoc} */
+        @Override public String toString() {
+            return "UserValue{" +
+                "x=" + x +
+                ", y=" + y +
+                ", z=" + z +
+                '}';
+        }
+    }
+
+    /**
+     * Cehcks that validate_indexes command will cancel after it interrupted.
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testValidateIndexesCommand() throws Exception {
+        lnsrLog = new ListeningTestLogger(false, log);
+
+        IgniteEx ignite = startGrid(0);
+
+        ignite.cluster().active(true);
+
+        IgniteCache cache = ignite(0).cache(DEFAULT_CACHE_NAME);
+
+        for (int i = 0; i < 500_000; i++)
+            cache.put(i, new UserValue(i));
+
+        LogListener lnsrValidationStarted = LogListener.matches("Current progress of ValidateIndexesClosure").build();
+        LogListener lnsrValidationCancelled = LogListener.matches("Index validation was cancelled.").build();
+
+        lnsrLog.registerListener(lnsrValidationStarted);
+        lnsrLog.registerListener(lnsrValidationCancelled);
+
+        IgniteInternalFuture fut = GridTestUtils.runAsync(() ->
+            assertNotSame(EXIT_CODE_OK, execute("--cache", "validate_indexes")));
+
+        assertTrue(GridTestUtils.waitForCondition(lnsrValidationStarted::check, 10_000));
+
+        fut.cancel();
+
+        fut.get();
+
+        assertTrue(GridTestUtils.waitForCondition(lnsrValidationCancelled::check, 10_000));
+    }
+
+    /**
+     * Test invokes index validation closure and canceling it after started.
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testCancelValidateIndexesClosure() throws Exception {
+        IgniteEx ignite0 = startGrid(0);
+
+        ignite0.cluster().active(true);
+
+        IgniteCache<Integer, UserValue> cache = ignite0.cache(DEFAULT_CACHE_NAME);
+
+        for (int i = 0; i < LOAD_LOOP; i++)
+            cache.put(i, new UserValue(i));
+
+        AtomicBoolean cancelled = new AtomicBoolean(false);
+
+        ValidateIndexesClosure clo = new ValidateIndexesClosure(cancelled::get, Collections.singleton(DEFAULT_CACHE_NAME),
+            0, 0, false, true);
+
+        ListeningTestLogger listeningLogger = new ListeningTestLogger(false, log);
+
+        GridTestUtils.setFieldValue(clo, "ignite", ignite0);
+        GridTestUtils.setFieldValue(clo, "log", listeningLogger);
+
+        LogListener lnsrValidationStarted = LogListener.matches("Current progress of ValidateIndexesClosure").build();
+
+        listeningLogger.registerListener(lnsrValidationStarted);
+
+        IgniteInternalFuture fut = GridTestUtils.runAsync(() -> {
+            try {
```

Comment:
Please use `GridTestUtils#assertThrows` instead.

Reply:
I don't know what it helps.

Reply:
It helps you avoid the code duplication and makes your code cleaner.

Reply:
Done.